### DISCUSSION TITLE
Keeping html tags when exporting to HTML

### DISCUSSION THREAD

**Discussion by marlarius** [dbeaver/dbeaver#22210] [Q&A]
When DBeaver exports to a HTML file, it converts all tags to their HTML entities. Thus, the content does not look right in a browser. Right now I have to edit the finished file and replace &lt; and &gt; etc. with their corresponding tags. 

Is there an option somewhere to NOT convert tags? Otherwise, there should be :) Preferably with an option to specify which tags should be converted, if any. Certain tags, like table tags, could mess up the layout. But even a simple checkbox would do.

**Comment by LonwoLonwo** [comment]
Hello @marlarius 

> Thus, the content does not look right in a browser. 

Could you please provide more info about this? How exactly it doesn't look right?

**Reply by marlarius** [reply]
MySQL Test table:
```
CREATE TABLE `mytable` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `html` varchar(100) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=3 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
```
Data:
`INSERT INTO mytable (html) VALUES ('line 1<br>line 2<br>line 3');`

Open the table data, right click and select Export data > HTML > Next x 4 > Proceed.

Open the exported table in a browser and you will get this output:
![image](https://github.com/dbeaver/dbeaver/assets/2758118/df838960-32c1-4116-ad6f-d3d22371cd00)

Whereas I would expect this where I have edited the html afterwards:
![image](https://github.com/dbeaver/dbeaver/assets/2758118/c17685b1-7fae-4b9b-bdd5-64c1f8652374)

The problem is caused by DBeaver converting special characters to their html entities, eg. &amp;gt; instead of > etc. In some cases that is indeed the wanted outcome, that's why it should be an option.

**Reply by ShadelessFox** [reply]
It's highly insecure to embed raw, unprocessed HTML data. It might contain a payload or code injection.

**Reply by marlarius** [reply]
@ShadelessFox: That is irrelevant for my question. I have full control of what is in the database. And we are talking about generating a flat HTML file for a report, so security is not an issue here. I don't say that you are wrong, but as always, it depends on the circumstances :)

**Comment by Sarveshwaran20** [comment]
DBeaver might not have a specific option to control tag conversion during HTML export. To address this issue, you may want to check the latest version of DBeaver or consult their official documentation or community forums for any updates or new features related to HTML export options. Keep in mind that software features can evolve over time, and it's possible that new options have been introduced since my last update.

**Comment by marlarius** [comment]
@Sarveshwaran20

Unfortunately there isn't anything to find in the official documentation, I already looked there.

You mention the community forums, but I am not able to find any other than this one. Could you please give me a link?

**Reply by Sarveshwaran20** [reply]
Did you check it out

**Comment by Sarveshwaran20** [comment]
DBeaver: Export data in HTML fomat is missing specification of character set to use #5058 


Try searching this issue above in Google or use this documentation this might help you
[Uploading dbeaver_v_22_2.pdf…]()

**Comment by Sarveshwaran20** [comment]
@marlarius did it work

**Comment by marlarius** [comment]
No, unfortunately still no luck. There are many options for exporting data, but apparently none that relate to how HTML tags are encoded.

BTW, the link to the PDF documentation is invalid but I found it elsewhere.

**Reply by Sarveshwaran20** [reply]
Ohh ok

**Reply by Sarveshwaran20** [reply]
DBeaver: Export data in HTML fomat is missing specification of character set to use #5058

**Reply by Sarveshwaran20** [reply]
This is something related to your issue check the issue I'd just might be your answer

**Reply by Sarveshwaran20** [reply]
Tell me if it works

**Comment by marlarius** [comment]
Thanks, I checked it out, but my problem is not related to specification of the character set. The problem is that certain characters like > and < are encoded to the html entities, like &gt; etc.

**Reply by Sarveshwaran20** [reply]
So you are trying to decode it ???

**Reply by Sarveshwaran20** [reply]
In DBeaver, you can use the `UNESCAPE` function to decode HTML entities. For example, if you want to decode `&lt;` to `<` and `&gt;` to `>`, you can use the following SQL query:

sql
SELECT UNESCAPE('your_encoded_string_here') AS decoded_string;


Replace 'your_encoded_string_here' with the actual encoded string you want to decode. This should help you handle HTML entities in DBeaver.

**Comment by marlarius** [comment]
There is no unescape function in Microsoft TSQL. 

But it also missed the point: the characters are NOT encoded, they are stored as plain text, like "< b >bold< /b >". But when DBeaver exports the table, THEN DBeaver encodes the characters.

Note: The space inside the tags are not in the table, it's only to avoid this forum to actually interpret the tags :)

**Reply by Sarveshwaran20** [reply]
If your characters are stored as plain text like "<b>bold</b>" and are being encoded upon export, you may need to disable HTML encoding or formatting options in DBeaver's export settings. Check the export configuration for any options related to encoding or formatting and adjust them accordingly.

**Comment by marlarius** [comment]
That's the problem in a nutshell: There are no such options!

**Reply by Sarveshwaran20** [reply]
https://www.google.com/url?q=https://github.com/dbeaver/cloudbeaver/wiki/Data-export%23:~:text%3DYou%2520can%2520perform%2520data%2520export,ResultSet%2520in%2520the%2520SQL%2520Editor.&sa=U&sqi=2&ved=2ahUKEwiLu-KjlemDAxXcSmwGHVWzBLEQFnoECBIQBQ&usg=AOvVaw1a1j8T2bCAqy6Ktb8UAnKn

**Reply by Sarveshwaran20** [reply]
feature request] useful export configuration #9728

**Comment by marlarius** [comment]
There are numerous options for other formats like csv but not for HTML. Nothing that relates to encoding of html tags.

![image](https://github.com/dbeaver/dbeaver/assets/2758118/a71234e2-fef9-47f2-b3f8-215d68c448b8)

**Reply by Sarveshwaran20** [reply]
If you're looking to decode HTML entities while formatting in DBeaver, you might want to use a function or expression to handle the decoding. Unfortunately, DBeaver's capabilities might depend on the specific database you are using.

As a general approach, you could look into functions like `html_entity_decode` or similar depending on your database. If you're dealing with a text column containing HTML entities, you might need to write a custom script or use a specific function provided by your database to decode these entities during retrieval.

If you provide more details about the database type and the specific scenario, I can try to guide you with a more precise solution.

**Comment by Sarveshwaran20** [comment]
If nothing works out just reach the dbeaver team and ask them to resolve the issue

**Comment by marlarius** [comment]
My approach now is to search and replace the corresponding tags in the file afterwards. It's a bit tedious, but it works. I had hoped there was a way to make DBeaver skip encoding the characters. I should perhaps make it a feature request instead.

**Reply by Sarveshwaran20** [reply]
Yep ok all the best replacing the corresponding tags .
